dashboard/test/models/pd/payment_terms_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'test_helper'
+
+class Pd::PaymentTermTest < ActiveSupport::TestCase
+  setup do
+    @regional_partner_1 = create :regional_partner
+    @regional_partner_2 = create :regional_partner
+
+    @workshop_1 = create :pd_workshop, num_sessions: 1, sessions_from: Date.new(2017, 1, 1), regional_partner: @regional_partner_1
+    @workshop_2 = create :pd_workshop, num_sessions: 1, sessions_from: Date.new(2017, 3, 1), regional_partner: @regional_partner_1
+    @workshop_3 = create :pd_workshop, regional_partner: @regional_partner_2
+  end
+
+  test 'throws exception if no applicable payment terms' do
+    create :pd_payment_term, regional_partner: @regional_partner_1, start_date: Date.new(2017, 4, 1)
+    [@workshop_1, @workshop_3].each do |workshop|
+      exception = assert_raises(Exception) do
```

Comment:
Don't use `Exception`. Use the more specific `RuntimeError`, which is the default for [raise](https://ruby-doc.org/core-2.4.1/Kernel.html#method-i-raise)